app/src/features/sessionBook/components/SaveSessionButton/SaveSessionButton.tsx

```diff
@@ -1,30 +1,50 @@
-import { useState } from "react";
-import { useSelector } from "react-redux";
+import React, { useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { getAppMode, getUserAttributes, getUserAuthDetails } from "store/selectors";
 import { RQButton } from "lib/design-system/components";
 import { MdOutlineFileDownload } from "@react-icons/all-files/md/MdOutlineFileDownload";
 import { SessionConfigPopup } from "features/sessionBook/screens/SavedSessionScreen/components/SessionConfigPopup/SessionConfigPopup";
 import DownArrow from "assets/icons/down-arrow.svg?react";
-import { useLocation } from "react-router-dom";
+import { useLocation, useNavigate } from "react-router-dom";
 import { CheckboxValueType } from "antd/lib/checkbox/Group";
 import { DebugInfo } from "features/sessionBook/types";
 import { getRecordingOptionsToSave } from "features/sessionBook/utils/sessionFile";
 import { downloadSessionFile } from "features/sessionBook/utils/sessionFile";
 import { getSessionRecordingEvents, getSessionRecordingMetaData } from "store/features/session-recording/selectors";
 import { toast } from "utils/Toast";
 import Logger from "lib/logger";
+import { SOURCE } from "modules/analytics/events/common/constants";
+import APP_CONSTANTS from "config/constants";
+import { saveDraftSession } from "features/sessionBook/screens/DraftSessionScreen/utils";
+import { getCurrentlyActiveWorkspace } from "store/features/teams/selectors";
+import { useIncentiveActions } from "features/incentivization/hooks";
+import { actions } from "store";
 import "./saveSessionButton.scss";
 
-export const SaveSessionButton = () => {
+interface SaveSessionButtonProps {
+  onSaveClick?: () => void;
+}
+
+export const SaveSessionButton: React.FC<SaveSessionButtonProps> = ({ onSaveClick }) => {
+  const dispatch = useDispatch();
   const location = useLocation();
+  const navigate = useNavigate();
+  const appMode = useSelector(getAppMode);
+  const user = useSelector(getUserAuthDetails);
+  const userAttributes = useSelector(getUserAttributes);
+  const workspace = useSelector(getCurrentlyActiveWorkspace);
 
   const sessionEvents = useSelector(getSessionRecordingEvents);
   const sessionRecordingMetadata = useSelector(getSessionRecordingMetaData);
 
   const [isPopupVisible, setIsPopupVisible] = useState(false);
   const isDraftSession = location.pathname.includes("draft");
   const debugInfoToBeIncluded: CheckboxValueType[] = [DebugInfo.INCLUDE_NETWORK_LOGS, DebugInfo.INCLUDE_CONSOLE_LOGS];
+  const [isLoading, setIsLoading] = useState(false);
+
+  const { claimIncentiveRewards } = useIncentiveActions();
```

Comment:
just saw that incentivisation does not have a check for `appFlavour` check. Please log a separate ticket for that to avoid sessionbear activity being counted towards giving incentives in requestly

Reply:
In case there is already a check and I missed it, ignore